### PR TITLE
Migrate to carpentries workbench

### PR DESCRIPTION
**Pull request by svenvanderburg**
This migrates the workshop to the carpentries workbench.
@reviewer: There are a lot of changed files, but I did not change anything in the content. I think the best way to test this is to render everything locally and see if something is wrong, see: [README](https://github.com/carpentries-incubator/deep-learning-intro/blob/57f26f1530926a750bafd51a784d0b1f67eecbba/README.md) for how to run it!

TODO:
* Make sure the website is deployed correctly. At least we should change the default branch to be `main` instead of `gh-pages`.

### CODE REVIEW

**Review by dsmits — COMMENTED** [review]
Could you add some instructions in the readme on running this stuff locally?

File: config.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# lc: Library Carpentry
+# cp: Carpentries (to use for instructor training for instance)
+# incubator: The Carpentries Incubator
+carpentry: 'incubator'
+
+# Overall title for pages.
+title: 'Introduction to deep learning'
+
+# Date the lesson was created (YYYY-MM-DD, this is empty by default)
+created: '2023-05-26'
```
Comment (dsmits) [inline config.yaml:17]: The lesson was first created like 3 years ago right?

File: episodes/2-keras.Rmd [hunk 2]
```diff
@@ -544,39 +532,39 @@ Here we only set a different number of `epochs`.
...
-> > 2. The results of the training give very little information on its performance on a test set.
-> >    You should be careful not to use it as an indication of a well trained network.
-> {:.solution}
-{:.challenge}
+::: challenge
+## The Training Curve
+Looking at the training curve we have just made.
+
+1. How does the training progress?
+ * Does the training loss increase or decrease?
```
Comment (dsmits) [inline episodes/2-keras.Rmd:556]: could you indent these bullet points?

File: learners/setup.md [hunk 3]
```diff
@@ -17,7 +18,13 @@ least a day in advance of the workshop.  If you encounter problems with the
...
 you are ready to go as soon as the workshop begins.
 
-### Windows - [Video tutorial][video-windows]
+:::::::::::::::::::::::::::::::::::::::::::::::::::
+
+:::::::::::::::: solution
+
+### Windows
+
+Checkout the [video tutorial][video-windows] or:
```
Comment (dsmits) [inline learners/setup.md:27]: These numbered lists are looking a bit awkward, could you format it differently? Either by putting the text on the same line as the number or making the number a little header

**Review by dsmits — CHANGES_REQUESTED** [review]
Just a few small comments. Looks good!

File: config.yaml (hunk 1 above)
Comment (svenvanderburg) [reply config.yaml]: Check I changed it to 2020-10-17

File: episodes/2-keras.Rmd (hunk 2 above)
Comment (svenvanderburg) [reply episodes/2-keras.Rmd]: Check

File: learners/setup.md (hunk 3 above)
Comment (svenvanderburg) [reply learners/setup.md:27]: Well-spotted! I fixed it :)

**Pull request merged**